File: php/slim-doctrine/app/src/com/sprint/sms/api/domain/Menu.php
<?php
namespace App\com\sprint\sms\api\domain;

use Doctrine\ORM\Mapping as ORM;

use \App\com\sprint\sms\api\base\domain\BaseEntryVersion;
use \App\com\sprint\sms\api\support\FormatSupport;

/**
 * @ORM\Entity
 * @ORM\Table(name = "t_menu")
 */
class Menu extends BaseEntryVersion implements FormatSupport
{

	/**
     * @ORM\Column(name = "ID_", type = "bigint")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy = "IDENTITY")
	 */
	private $id;
	
	/**
     * @ORM\Column(name = "title", type = "string", length = 150, nullable = false)
     */
    private $title;
	
	/**
     * @ORM\Column(name = "link", type = "string", nullable = true)
     */
    private $link;
	
	/**
     * @ORM\Column(name = "icon", type = "string", nullable = true)
     */
    private $icon;
	
	/**
     * @ORM\Column(name = "description", type = "string", nullable = true)
     */
    private $description;
	
	/**
     * @ORM\Column(name = "active", type = "boolean", nullable = false, options = {"default":true})
     */
    private $active;
	
	/**
     * //ORM\ManyToOne(targetEntity = "Menu", inversedBy = "children")
	 * @ORM\ManyToOne(targetEntity = "Menu")
     * @ORM\JoinColumn(name = "f_parent", referencedColumnName = "ID_", onDelete = "SET NULL", nullable = true)
     */
    private $parent;
	
	/**
     * @ORM\Column(name = "priority", type = "bigint", nullable = false)
     */
    private $priority;
	
	/**
     * @ORM\Column(name = "global", type = "boolean", nullable = false, options = {"default":false})
     */
    private $global;
	
	
	private $children;
	
	private $action;
	
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}
	
	public function setTitle($title) {
		$this->title = $title;
	}
	
	public function getTitle() {
		return $this->title;
	}	
	
	public function setLink($link) {
		$this->link = $link;
	}
	
	public function getLink() {
		return $this->link;
	}
	
	public function setIcon($icon) {
		$this->icon = $icon;
	}
	
	public function getIcon() {
		return $this->icon;
	}
	
	public function setDescription($description) {
		$this->description = $description;
	}
	
	public function getDescription() {
		return $this->description;
	}
	
	public function setActive($active) {
		$this->active = $active;
	}
	
	public function getActive() {
		return $this->active;
	}
	
	public function setParent(Menu $parent = null) {
		$this->parent = $parent;
	}
	
	public function getParent() {
		return $this->parent;
	}
	
	public function setPriority($priority) {
		$this->priority = $priority;
	}
	
	public function getPriority() {
		return $this->priority;
	}
	
	public function setGlobal($global) {
		$this->global = $global;
	}
	
	public function getGlobal() {
		return $this->global;
	}	
	
	public function setChildren($children) {
		$this->children = $children;
	}
	
	public function getChildren() {
		return $this->children;
	}	
	
	public function setAction(array $action = null) {
		$this->action = $action;
	}
	
	public function getAction() {
		return $this->action;
	}
	
	public function toFormatObject() {
		$o = parent::toFormatObject();
		$o["id"] = $this->id;
		$o["title"] = $this->title;
		$o["link"] = $this->link;	
		$o["icon"] = $this->icon;	
		$o["description"] = $this->description;	
		$o["active"] = $this->active ? 1 : 0;	
		$o["parent"] = $this->parent !== null ? array("id" => $this->parent->getId()) : null;	
		$o["priority"] = $this->priority;	
		$o["global"] = $this->global ? 1 : 0;	
		$o["action"] = $this->action;		
		if (isset($this->children)) {
			$o["children"] = array();
			for ($i = 0; $i < count($this->children); $i++) {
				$c[Menu::VERSION] = $this->children[$i]->getVersion();
				$c[Menu::ENTRY] = $this->children[$i]->getEntryTime() !== null ? $this->children[$i]->getEntryTime()->getTimestamp() * 1000 : null;
				$c["id"] = $this->children[$i]->id;
				$c["title"] = $this->children[$i]->title;
				$c["link"] = $this->children[$i]->link;	
				$c["icon"] = $this->children[$i]->icon;	
				$c["description"] = $this->children[$i]->description;	
				$c["active"] = $this->children[$i]->active ? 1 : 0;	
				$c["parent"] = $this->children[$i]->parent !== null ? array("id" => $this->children[$i]->parent->getId()) : null;	
				$c["priority"] = $this->children[$i]->priority;	
				$c["global"] = $this->children[$i]->global ? 1 : 0;	
				$c["action"] = $this->children[$i]->action;
				$o["children"][$i] = $c;
			}
		}		
		return $o;
	}
	
}

File: java/spring-hibernate/src/main/java/com/sprint/sms/api/domain/Menu.java
package com.sprint.sms.api.domain;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Transient;

import com.ideahut.common2.annotation.IdhFormatter;
import com.ideahut.shared2.domain.base.LongIdEntryVersionDomain;

@Entity
@Table(name = "t_menu")
@IdhFormatter
public class Menu extends LongIdEntryVersionDomain {	
	/**
	 * 
	 */
	private static final long serialVersionUID = 5611248316336571900L;

	private String title;
	
	private String link;
	
	private String icon;
	
	private String description;
	
	private Boolean active;
	
	private Menu parent;
	
	private Long priority; 
	
	private Boolean global;
	
	private List<Menu> children = new ArrayList<Menu>();
	
	private Set<String> action = new HashSet<String>();
	

	@Column(name = "title", length = 150, nullable = false)
	@IdhFormatter
	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	@Column(name = "link")
	@IdhFormatter
	public String getLink() {
		return link;
	}

	public void setLink(String link) {
		this.link = link;
	}

	@Column(name = "icon")
	@IdhFormatter
	public String getIcon() {
		return icon;
	}

	public void setIcon(String icon) {
		this.icon = icon;
	}
	
	@Column(name = "description")
	@IdhFormatter
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	@Column(name = "active", nullable = false)
	@IdhFormatter
	public Boolean getActive() {
		return active;
	}

	public void setActive(Boolean active) {
		this.active = active;
	}

	@ManyToOne
	@JoinColumn(name = "f_parent")
	@IdhFormatter(field = Menu.ID)
	public Menu getParent() {
		return parent;
	}

	public void setParent(Menu parent) {
		this.parent = parent;
	}

	@Column(name = "priority", nullable = false)
	@IdhFormatter
	public Long getPriority() {
		return priority;
	}

	public void setPriority(Long priority) {
		this.priority = priority;
	}
	
	@Column(name = "global", nullable = false)
	@IdhFormatter
	public Boolean getGlobal() {
		return global;
	}

	public void setGlobal(Boolean global) {
		this.global = global;
	}

	@Transient
	@IdhFormatter
	public List<Menu> getChildren() {
		return children;
	}

	public void setChildren(List<Menu> children) {
		this.children = children;
	}
	
	@Transient
	@IdhFormatter
	public Set<String> getAction() {
		return action;
	}

	public void setAction(Set<String> action) {
		this.action = action;
	}
	
}
